src/modeldata.jl

```diff
@@ -75,9 +75,48 @@ TransformerWeights(p::Config) = TransformerWeights(;
     rms_final_weight = zeros(Float32, p.dim),
     freq_cis_real = zeros(Float32, (p.dim ÷ p.n_heads) ÷ 2, p.seq_len),
     freq_cis_imag = zeros(Float32, (p.dim ÷ p.n_heads) ÷ 2, p.seq_len),
-    wcls = zeros(Float32, p.dim, p.vocab_size) 
+    wcls = zeros(Float32, p.dim, p.vocab_size)
 )
 
+@views function mmap_weights(p::Config, checkpoint_filename)
+    v = Mmap.mmap(checkpoint_filename)
+    # Drop first 7*4=28 bytes that correspond to Config
+    f = reinterpret(Float32, v)[8:end]
+
+    marker = 1
+
+    load = (n...) -> begin
```

Comment:
lgtm; would like to figure out if we can reuse the same `load` interface for the ordinary in-core loader. In principle, there should be no different to read out a stream of objects from some iostream, whether or not it's a mmap or regular file?

Reply:
I gave this a go in fff09a380a5c7b289eb2e0401002f20afc1e52b7 by unconditionally `mmap`ing then optionally materializing the arrays. I don't think this is quite what you were after here though.

One thing to note is CSV.jl unconditionally mmap's to read files so it makes me think it's not a crazy thing to do

Reply:
I remember now why I couldn't re-use the existing `load` code: because with `read!` we need to read into a preallocated object that already exists. The marker + `view`-based code let's us instead build the layers directly as views into the `mmap`'d data. I believe this makes a big difference with the large models.